### PR TITLE
Rainbow local mode

### PR DESCRIPTION
**Pull request by assapin**
* docker compose
* packaging script. the output of this script is a .zip file with 
- rainbow_dags.py in the root
- the rest of the rainbow package as folders
- all the dependencies of rainbow which don't already come with airflow in the root of the zip
* RAINBOW_HOME is where the build puts the script
* this is mounted also to the docker to load automaticall
* the yamls are assumed to be by default in RAINBOW_HOME/rainbows
* ability to override the Variable.get for local runs
* other stuff to do with class loading which didn't work in the zip - fixed in class utils

important:
the tests still fail for me since the aws_hook is activated by default and expects a connection to cloudwatch.
I didn't have time to fix it, but I think we should add the ability to turn it off

### CODE REVIEW

File: rainbow/core/util/class_util.py [hunk 1]
```diff
@@ -19,40 +19,47 @@
...
-                                module_name = mod.__name__
-                                class_name = obj.__name__
-                                parent_module = root[len(py_path) + 1:].replace('/', '.')
-                                module = parent_module.replace('airflow.dags.', '') + \
-                                         '.' + module_name
-                                clazz = __get_class(module, class_name)
-                                if issubclass(clazz, parent_class):
-                                    classes.update({module_name: clazz})
-    return classes
+    module_content = {}
```
Comment (aviem-naturalint) [inline rainbow/core/util/class_util.py:33]: This code change introduces failure in test 
```python -m unittest tests.runners.airflow.build.test_build_rainbows.TestBuildRainbows```

File: README.md [hunk 2]
```diff
@@ -84,4 +84,24 @@ https://github.com/Natural-Intelligence/rainbow/tree/master/tests/runners/airflo
...
 
-TODO: installation.
+* clone the repository
+* set an environment variable RAINBOW_HOME to point to a local folder outside the git repository
+* from the root of the project, run ./rainbow-cli package_dags $RAINBOW_HOME
+this will create a rainbows.zip file in your RAINBOW_HOME
+* place a rainbow yaml file in RAINBOW_HOME/rainbows
+* edit deploy/local/.env to point to your RAINBOW_HOME
+* make sure you have docker running on your local machine
+* docker-compose -f deploy/local/docker-compose.yaml up
```
Comment (aviem-naturalint) [inline README.md:94]:
```suggestion
* docker-compose -f deploy/local/docker-compose.yml up
```

File: README.md [hunk 3]
```diff
@@ -84,4 +84,24 @@ https://github.com/Natural-Intelligence/rainbow/tree/master/tests/runners/airflo
 
 # Installation
 
-TODO: installation.
+* clone the repository
+* set an environment variable RAINBOW_HOME to point to a local folder outside the git repository
+* from the root of the project, run ./rainbow-cli package_dags $RAINBOW_HOME
```
Comment (aviem-naturalint) [inline README.md:89]:
```suggestion
* from the root of the project, run ./rainbow-cli package-dags $RAINBOW_HOME
```

File: README.md (hunk 2 above)
Comment (aviem-naturalint) [reply README.md:94]: Please add this as a command in the cli, something like `./rainbow-cli start`

File: README.md (hunk 3 above)
Comment (aviem-naturalint) [reply README.md:89]: Should this even be a command in the cli?
It just runs `package.sh`
If this is for rainbow developers, we can instead instruct them to use `package.sh`

File: deploy/local/.env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+RAINBOW_HOME=/Users/assaf/projects/rainbow_home
```
[the remark this answers is not in the document]
Comment (assapin) [reply deploy/local/.env:1]: done

File: rainbow/runners/airflow/config/standalone_variable_backend.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+def get_variable(key, default_val):
+    if rainbow_local_mode():
+        result = EnvironmentVariablesBackend().get_variable(key) or default_val
+    else:
+        return Variable.get(key, default_var=default_val)
+
+
+def rainbow_local_mode():
+    return distutils.util.strtobool(environ.get(RAINBOW_STAND_ALONE_MODE_KEY, "False"))
```
[the remark this answers is not in the document]
Comment (assapin) [reply rainbow/runners/airflow/config/standalone_variable_backend.py:18]: [AIRFLOW-3876] AttributeError: module ‘distutils’ has no attribute ‘util’
removed it. sucks

File: rainbow/build/build_rainbows.py [hunk 6]
```diff
@@ -74,33 +74,38 @@ def __build_image(base_path, builder_config, builder):
...
+#user_image_builders_package = 'TODO: user_image_builders_package'
 
 task_build_classes = class_util.find_subclasses_in_packages(
-    [image_builders_package, user_image_builders_package],
+    [image_builders_package],
     ImageBuilder)
 
+
+
+task_build_types = {x.split(".")[-2]:c for x,c in task_build_classes.items()}
```
Comment (aviem-naturalint) [inline rainbow/build/build_rainbows.py:96]: Can we have a helper method for this with a descriptive name?

**Pull request merged**